test/e2e/k8s_util.go

```diff
@@ -728,26 +727,6 @@ func (data *TestData) DeleteTier(name string) error {
 	return nil
 }
 
-// CreateOrUpdateV1Alpha2CG is a convenience function for idempotent setup of crd/v1alpha2 ClusterGroups
-func (data *TestData) CreateOrUpdateV1Alpha2CG(cg *crdv1alpha2.ClusterGroup) (*crdv1alpha2.ClusterGroup, error) {
-	log.Infof("Creating/updating ClusterGroup %s", cg.Name)
-	cgReturned, err := data.crdClient.CrdV1alpha2().ClusterGroups().Get(context.TODO(), cg.Name, metav1.GetOptions{})
-	if err != nil {
-		cgr, err := data.crdClient.CrdV1alpha2().ClusterGroups().Create(context.TODO(), cg, metav1.CreateOptions{})
-		if err != nil {
-			log.Infof("Unable to create cluster group %s: %v", cg.Name, err)
-			return nil, err
-		}
-		return cgr, nil
-	} else if cgReturned.Name != "" {
-		log.Debugf("ClusterGroup with name %s already exists, updating", cg.Name)
-		cgReturned.Spec = cg.Spec
-		cgr, err := data.crdClient.CrdV1alpha2().ClusterGroups().Update(context.TODO(), cgReturned, metav1.UpdateOptions{})
-		return cgr, err
-	}
-	return nil, fmt.Errorf("error occurred in creating/updating ClusterGroup %s", cg.Name)
-}
-
 // CreateOrUpdateV1Alpha3CG is a convenience function for idempotent setup of crd/v1alpha3 ClusterGroups
 func (data *TestData) CreateOrUpdateV1Alpha3CG(cg *crdv1alpha3.ClusterGroup) (*crdv1alpha3.ClusterGroup, error) {
```

Comment:
Shall we just rename it to `CreateOrUpdateClusterGroup` then (same for the delete and builder functions), otherwise when API is graduated to v1beta this will be confusing.

Reply:
Works for me. If v1beta1 is not backwards-compatible with v1alpha3, then we can always introduce the version suffix back.

Reply:
Actually, I'd rather not change it in this PR. I see that we have the same issue for the `Group` CRD, even though we ever only had one version for that one. We should probably handle them together as a separate PR, or when we introduce v1beta1.

Reply:
Okay, I'll take note on this one